lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * @fileoverview Rule to flag numbers that will lose significant figure precision at runtime
+ * @author Jacob Moore
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        type: "problem",
+
+        docs: {
+            description: "disallow literal numbers that lose precision",
+            category: "Possible Errors",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/no-loss-of-precision"
+        },
+        schema: [],
+        messages: {
+            noLossOfPrecision: "This number literal will lose precision at runtime."
+        }
+    },
+
+    create(context) {
+
+        /**
+         * Returns whether the node is number literal
+         * @param {Node} node the node literal being evaluated
+         * @returns {boolean} true if the node is a number literal
+         */
+        function isNumber(node) {
+            return typeof node.value === "number";
+        }
+
+
+        /**
+         * Returns whether the string only contains octal digits
+         * @param {string} rawString the string representation of the number being evaluated
+         * @returns {boolean} true if the string contains only octal digits
+         */
+        function isOctalDigitsOnly(rawString) {
+            return rawString.split("").every(digit => parseInt(digit, 10) < 8);
+        }
+
+        /**
+         * Checks whether the number is  base ten
+         * @param {ASTNode} node the node being evaluated
+         * @returns {boolean} true if the node is in base ten
+         */
+        function isBaseTen(node) {
+            const prefixes = ["0x", "0X", "0b", "0B", "0o", "0O"];
+
+            return prefixes.every(prefix => !node.raw.startsWith(prefix)) &&
```

Comment:
I think this will treat the literal `0` as octal, which is incorrect although it doesn't affect the result of the precision check.